### PR TITLE
Generic V3 improvements

### PR DESCRIPTION
**Pull request by denysvitali**
This PR adds the support for multiple V3 endpoints by adding the following methods:

### Apps

- `SetV3AppEnvVariables`

### Domains

- `ListV3Domains`

### Packages

- `CreateV3DockerPackage`

### Routes

- `ListV3Route`
- `CreateV3Route`

### Service Instances

- `ListV3ServiceInstances`
- `ListV3ServiceInstancesByQuery`

---

Since this is quite a big PR, I've tried to split it in #300, and #301, but I guess it's easier just to merge this bigger `v3-improvements` PR instead.

### CODE REVIEW

File: v3apps.go [hunk 1]
```diff
@@ -255,3 +255,31 @@ func (c *Client) GetV3AppEnvironment(appGUID string) (V3AppEnvironment, error) {
 
 	return result, nil
 }
+
+type V3EnvVar struct {
+	Var map[string]*string `json:"var"`
```
Comment (sneal) [inline v3apps.go:260]: That's unfortunate the value is a pointer to string, but unfortunately this is technically correct according to the API docs.

**Comment by sneal** [comment]
Thanks for the contribution! Works on my machine.

**Comment by denysvitali** [comment]
> Works on my machine.

We'll ship your machine then 😂

**Comment by sneal** [comment]
@denysvitali Just so you are aware, there was a `V3ToManyRelationship` merge conflict from another PR that I fixed.

**Pull request merged**